workflow-dev/tpl/storage.sh

```diff
@@ -1,9 +1,7 @@
 #!/bin/bash
 store=`grep "storage\s*=\s*" $HELM_GENERATE_DIR/tpl/generate_params.toml | cut -d "=" -f2 | tr -dc '[:alnum:]\n\r'`
-if [ -n "$STORAGE_TYPE" ]; then
-  if [ "$STORAGE_TYPE" != "minio" ]; then
-    rm -rf $HELM_GENERATE_DIR/manifests/deis-minio-*
-  fi
+if [ -n "$STORAGE_TYPE" ] && [ "$STORAGE_TYPE" != "minio" ]; then
+  rm -rf $HELM_GENERATE_DIR/manifests/deis-minio-*
 elif [ "$store" != "minio" ]; then
   rm -rf $HELM_GENERATE_DIR/manifests/deis-minio-*
 else
```

Comment:
Note on positioning of this comment... I cannot place it any further down (past the diff), but I mean for this to apply to the _entire_ `if / elif / else / fi` block.

This logic can all be expressed more succinctly and with less confusion. Currently, there are two paths to the same result. (That result being `rm -rf $HELM_GENERATE_DIR/manifests/deis-minio-*`.) Also, currently, there's a lot of "negative logic." So for a human reading the code, it's backwards from how people usually think. It's starting with "if something isn't true; otherwise..." instead of starting out with "if something _is_ true; otherwise..."

I would suggest the following:

```
if [ "$STORAGE_TYPE" = "minio" ] || [ "$store" != "minio" ]; then
  helmc tpl -d $HELM_GENERATE_DIR/tpl/generate_params.toml -o $HELM_GENERATE_DIR/manifests/deis-minio-rc.yaml $HELM_GENERATE_DIR/tpl/deis-minio-rc.yaml
  cp $HELM_GENERATE_DIR/tpl/deis-minio-service* $HELM_GENERATE_DIR/manifests/
else
  rm -rf $HELM_GENERATE_DIR/manifests/deis-minio-*
fi
```